crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -349,34 +386,33 @@ impl<'w> EntityMut<'w> {
         let new_archetype = &mut archetypes[new_archetype_id];
 
         let new_location = if old_table_id == new_archetype.table_id() {
-            unsafe { new_archetype.allocate(entity, old_table_row) }
+            new_archetype.allocate(entity, old_table_row)
         } else {
             let (old_table, new_table) = storages
                 .tables
                 .get_2_mut(old_table_id, new_archetype.table_id());
 
-            // SAFE: table_row exists. All "missing" components have been extracted into the bundle
-            // above and the caller takes ownership
-            let move_result =
-                unsafe { old_table.move_to_and_forget_missing_unchecked(old_table_row, new_table) };
+            // SAFE: old_table_row exists
+            let move_result = match DROP {
```

Comment:
Is there a good reason to use match over if statements here? I'd prefer to not suppress clippy lints for this logic.

Reply:
Mostly just a style thing, match takes up less lines and I find it clearer, i can change it if you'd like

Reply:
I respect your preference, but I'll make counter arguments because we aren't real programmers if we don't debate trivial style changes:

* match might be fewer lines, but it is way less terse (match: 204 characters, if/else: 191 characters)
* consistency with the rest of the rust ecosystem is good
* suppressing clippy errors isn't fun
* if we can't use if / else for bools, why have them at all? Are we still programmers if we don't have if/else in our toolset?